src/ZipFile.jl

```diff
@@ -408,11 +416,12 @@ function close(f::WritableFile)
     f.compressedsize = position(f)
 
     # fill in local file header fillers
+    pos = position(f._io)
     seek(f._io, f._offset+14)   # seek to CRC-32
     _writele(f._io, UInt32(f.crc32))
     _writele(f._io, UInt32(f.compressedsize))
     _writele(f._io, UInt32(f.uncompressedsize))
-    seekend(f._io)
+    seek(f._io, pos)
```

Comment:
I'm guessing this is necessary for the case when an appended file entry is smaller than the previously written central directory? This is nicer anyway.